gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreBranch.java

```diff
@@ -52,36 +53,30 @@ public String getName() {
 
   public abstract String getBranchTypeString(boolean capitalized);
 
-  @SuppressWarnings("regexp")
+  @SuppressWarnings("regexp") // to allow `synchronized`
   @Override
   public synchronized GitCoreCommit getPointedCommit() throws GitCoreException {
     if (pointedCommit == null) {
       var revStr = getFullName();
-      var revCommit = resolveRevCommit(revStr);
+      @Unique RevCommit revCommit = resolveRevCommit(revStr);
       pointedCommit = new GitCoreCommit(revCommit);
     }
     return pointedCommit;
   }
 
-  /**
-   * - - - IMPORTANT NOTE - - -
-   * Bear in mind that RevCommit is a mutable object.
-   * Its internal state (inDegree, flags) changes during a rev walk (among others).
-   * To avoid potential bugs:
-   * - reinstance instead of reuse; this method provides "clean" instance based on a String revision
-   * - narrow the scope where a RevCommit is available; a use as a field is strongly discouraged
-   * This comment applies everywhere.
-   *
-   * @param revStr revision String
-   *
-   * @throws GitCoreException unable to resolve the revision
-   *
-   * @return {@link RevCommit} specified by {@code revStr}
-   */
-  protected RevCommit resolveRevCommit(String revStr) throws GitCoreException {
+  // - - - IMPORTANT NOTE - - -
+  // Bear in mind that RevCommit is a mutable object.
+  // Its internal state (inDegree, flags) changes during a rev walk (among others).
+  // To avoid potential bugs:
+  // - reinstantiate instead of reuse; this method provides a "clean" instance based on the given String revision
+  // - narrow the scope where a RevCommit is available; a use as a field is strongly discouraged.
+  // This comment applies everywhere in the codebase.
+  // Note that both points are kind-of enforced by Checkstyle (every occurrence of "RevCommit" must be preceded with Checker's @Unique annotation),
```

Comment:
Btw I've just added https://github.com/typetools/checker-framework/issues/3313 (`Feature request: @AlwaysUnique (?) to mark classes whose instances must always be @Unique`)